CRM/Event/BAO/Event.php

```diff
@@ -2063,31 +2063,94 @@ public static function checkRegistration($params) {
    * @param int $eventId
    * @param int $type
    *
-   * @return string
-   *   the permission that the user has (or null)
+   * @return bool|array
+   *   Whether the user has permission for this event (or if eventId=NULL an array of permissions)
+   * @throws \CiviCRM_API3_Exception
    */
   public static function checkPermission($eventId = NULL, $type = CRM_Core_Permission::VIEW) {
+    if (empty($eventId)) {
+      return self::getAllPermissions($type);
+    }
+
+    if (!isset(Civi::$statics[__CLASS__]['permission'][$eventId])) {
+      Civi::$statics[__CLASS__]['permission'][$eventId] = FALSE;
+
+      $params = array(
+        'id' => $eventId,
+        'check_permissions' => 1,
+        'return' => 'id, created_id',
+      );
+
+      $allEvents = [];
+      $createdEvents = [];
+      $contactId = CRM_Core_Session::getLoggedInContactID();
+      $eventResult = civicrm_api3('Event', 'get', $params);
+      if ($eventResult['count'] > 0) {
+        foreach ($eventResult['values'] as $eventId => $eventDetail) {
+          $allEvents[$eventId] = $eventId;
+          if (isset($eventDetails['created_id']) && $contactId == $eventDetail['created_id']) {
+            $createdEvents[$eventId] = $eventId;
+          }
+        }
+      }
+
+      // Note: for a multisite setup, a user with edit all events, can edit all events
+      // including those from other sites
+      if (($type == CRM_Core_Permission::EDIT) && CRM_Core_Permission::check('edit all events')) {
+        Civi::$statics[__CLASS__]['permission'][$eventId] = TRUE;
```

Comment:
I think you need an action nuance in your keys? ie 

Civi::$statics[__CLASS__]['permission']['edit'][$eventId] = TRUE;
Civi::$statics[__CLASS__]['permission']['view'][$eventId] = TRUE;

(set both on edit perm, only one on view )

Reply:
No. It's one of the magical mystery functions that returns different things depending on what you feed it.